Relax parsing rules for CSV translation files

nolandc opened the pull request:
Fixes https://github.com/godotengine/godot/issues/82002

This PR relaxes the parsing rules in the CSV translation importer, so that CSVs that are not properly formatted don't fail the import. This allows users to change the delimiter options (or set the mode to "keep"). 

More specifically, this would no longer throw an error for:
- CSVs that don't have more than 1 line (this could happen if a non-comma delimiter is set)
- CSVs that have keys without the expected number of translations (this could happen if a non-comma delimiter is set but a comma is used elsewhere in the file)

Testing this is a bit tedious at the moment due a crash in import_dock.cpp when reimporting files that don't return a resource from `ResourceLoader::load`. Fix for that is up here: https://github.com/godotengine/godot/pull/82490

nolandc committed:
warn instead of failing on malformed csv import

add translation tests

Scoppio commented:
There are no chances of adding unit tests for this? Because this can totally be a thing to be unit tested without too much trouble.

NolanDC commented:
Good idea, I'll add some tests!

NolanDC commented:
Added tests for:
- importing a file that has a malformed header row (imports with no data)
- importing a file that has a malformed translation row (skips importing the line in question)

I also moved the existing csv import test to this new file. I don't think it belongs in the test_translation.h file.

There didn't seem to be much precedent for these tests, as none of the other importers have tests. It seemed like it might be a good idea to have a folder for things that test pure editor code, as opposed to being shoved into existing files with a bunch of TOOLS_ENABLED conditionals. Then the test files can just be included with a single conditional. In any case, let me know if there's a better way to organize this and I'll update as needed.

Calinou reviewed (commented):
The reasoning behind this makes sense to me.

However, recent commits have added uses of `ERR_CONTINUE_MSG` instead of using `WARN_PRINT`. When you have time, could you rebase this PR on `master` and change your added code to use `ERR_CONTINUE_MSG` instead? Note that 4.3's release is imminent, so you don't have to rush for this as this PR is now targeting 4.4 (although it could be cherry-picked to 4.3.x later).
